kythe/go/util/keys/keys.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package keys implements orderedcode encodings for Kythe types.
+package keys
+
+import (
+	"github.com/google/orderedcode"
+
+	spb "kythe.io/kythe/proto/storage_go_proto"
+)
+
+// Append appends the encoded representations of items to the key buffer.
+//
+// In addition to the types supported by the orderedcode library, the following
+// Kythe types can be handled:
+//
+//  - *spb.VName
+//  - int32/int
+//
+// More detail at: https://godoc.org/github.com/google/orderedcode#Append
+func Append(key []byte, items ...interface{}) ([]byte, error) {
+	expanded := make([]interface{}, 0, len(items))
+	for _, item := range items {
+		switch x := item.(type) {
+		case *spb.VName:
+			expanded = append(expanded, x.Corpus, x.Language, x.Path, x.Root, x.Signature)
+		case int32:
```

Comment:
worth a comment about why forcing to int64?  I dunno.  Maybe I'm missing context.

Reply:
✓